src/com/goide/inspections/GoPlaceholderChecker.java

```diff
@@ -0,0 +1,497 @@
+
+  @NotNull private static List<PrintVerb> printVerbs = ContainerUtil.newArrayList(
+    new PrintVerb('%', "", 0),
+    new PrintVerb('b', " -+.0", argInt.getValue() | argFloat.getValue() | argComplex.getValue()),
+    new PrintVerb('c', "-", argRune.getValue() | argInt.getValue()),
+    new PrintVerb('d', " -+.0", argInt.getValue()),
+    new PrintVerb('e', " -+.0", argFloat.getValue() | argComplex.getValue()),
+    new PrintVerb('E', " -+.0", argFloat.getValue() | argComplex.getValue()),
+    new PrintVerb('f', " -+.0", argFloat.getValue() | argComplex.getValue()),
+    new PrintVerb('F', " -+.0", argFloat.getValue() | argComplex.getValue()),
+    new PrintVerb('g', " -+.0", argFloat.getValue() | argComplex.getValue()),
+    new PrintVerb('G', " -+.0", argFloat.getValue() | argComplex.getValue()),
+    new PrintVerb('o', " -+.0#", argInt.getValue()),
+    new PrintVerb('p', "-#", argPointer.getValue()),
+    new PrintVerb('q', " -+.0#", argRune.getValue() | argInt.getValue() | argString.getValue()),
+    new PrintVerb('s', " -+.0", argString.getValue()),
+    new PrintVerb('t', "-", argBool.getValue()),
+    new PrintVerb('T', "-", anyType.getValue()),
+    new PrintVerb('U', "-#", argRune.getValue() | argInt.getValue()),
+    new PrintVerb('v', " -+.0#", anyType.getValue()),
+    new PrintVerb('x', " -+.0#", argRune.getValue() | argInt.getValue() | argString.getValue()),
+    new PrintVerb('X', " -+.0#", argRune.getValue() | argInt.getValue() | argString.getValue())
+  );
+
+  private boolean argCanBeChecked(@NotNull GoExpression placeholder, int callArgs, int formatArg, @NotNull FormatState state) {
+    int argNum = state.argNums.get(formatArg);
+    if (argNum < 0) return false;
+
+    if (argNum == 0) {
+      String message = String.format("Index value [0] for <code>#ref</code>(\"%s\"); indexes start at 1", state.format);
+      myHolder.registerProblem(placeholder, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+      return false;
+    }
+
+    if (argNum < callArgs - 1) return true;
+    if (hasVariadic(myCallExpr.getArgumentList())) return false;
+    if (argNum < callArgs) return true;
+
+    // There are bad indexes in the format or there are fewer arguments than the format needs
+    // This is the argument number relative to the format: Printf("%s", "hi") will give 1 for the "hi"
+    int arg = argNum - state.firstArg + 1;
+    String message = String.format("Got %d placeholder(s) for %d arguments(s)", arg, callArgs - state.firstArg);
+    myHolder.registerProblem(placeholder, message, ProblemHighlightType.GENERIC_ERROR_OR_WARNING);
+
+    return false;
+  }
+
+  private static boolean isFunctionValue(GoExpression expression) {
+    return expression instanceof GoCallExpr;
+  }
+
+  private boolean okPrintfArg(@NotNull GoExpression placeholder, @NotNull List<GoExpression> arguments, @NotNull FormatState state) {
+    PrintVerb v = null;
+    boolean found = false;
+
+    for (int i = 0; i < printVerbs.size(); i++) {
```

Comment:
`found` variable is completely redundant: if you cannot find `verb` the `v` variable should be null.
whole loop looks silly. just replace printVerbs with any map.